pymc/data.py

```diff
@@ -499,17 +505,22 @@ def determine_coords(model, value, dims: Optional[Sequence[str]] = None) -> Dict
             )
         for size, dim in zip(value.shape, dims):
             coord = model.coords.get(dim, None)
-            if coord is None:
+            if coord is None and dim is not None:
```

Comment:
I don't have the full context of this function but in case it's appropriate, can we also take advantage of static shape of Aesara`s variables? Every entry in `tensorvariable.type.shape` should either be `None` or an integer for the size of that dimension.

Reply:
The context of this function is to extract coords from data variales such as `pd.Series` or `pd.DataFrame` that have an index.
I extended the signature to also extract `dims`, because indexes can be named.

It would also be neat to consider `xarray` variables here, ebcause they are a 1:1 match for `pm.Data`.

The `and dim is not None` I added is because `dims` elements can be `None` and then the `coords[None]` doesn't make sense.

What you're talking about with the static shapes might be more relevant in the `pm.Data` function itself, where I created the local `xshape` variable.